### PR TITLE
Rebuilds Luxury Shuttle (mostly), makes it emag-only

### PR DESCRIPTION
**Pull request by mc-oofert**
## About The Pull Request

Changes the really goofy old lux shuttle to a cooler layout with some additions to make it a luxury and not just
"anti-poor-people protection + food"

Shuttle was made bigger to make it less cramped for the luxury class, pool was moved to its own room, added an arcade
and a bar corner, has real lasers to shoot poors with (20c each shot), has fire extinguishers now
Adds a new preopen variant of hardened shutters
Adds a paywall pin subtype for the luxury shuttle, and a laser gun subtype

Made emag-only at a price of 10000 credits
## Why It's Good For The Game

The old luxury shuttle looked REALLY awful with its pool, was pretty cramped even in the luxury section and BARELY resembled a luxury..
This luxury shuttle provides luxuries such as a less poorly designed pool, more space for legs, arcade, to make it resemble a luxury unlike the old one

## Changelog
:cl:
add: Luxury Shuttle is now bigger, and less ugly! Poor people still get it rough though...
/:cl:

### CODE REVIEW

File: code/modules/shuttle/shuttle_rotate.dm [hunk 1]
```diff
@@ -110,3 +110,8 @@ If ever any of these procs are useful for non-shuttles, rename it to proc/rotate
 	. = ..()
 	if(wall_turret_direction && (params & ROTATE_DIR))
 		wall_turret_direction = turn(wall_turret_direction,rotation)
+
+//shutters that go from the sides look bad
+/obj/machinery/door/poddoor/shutters/shuttleRotate(rotation, params)
+	. = ..()
+	params = NONE
```
Comment (ZephyrTFA) [inline code/modules/shuttle/shuttle_rotate.dm:117]: This doesn't do anything.

Comment (mc-oofert) [reply code/modules/shuttle/shuttle_rotate.dm:117]: i swear it worked when i tried

**Comment by unit0016** [comment]
Why remove shuttle rotation on the shutters instead of just... rotating the shutters?  Would definitely shaft other shuttles where these are rotated, especially with wallening in the picture.

File: _maps/shuttles/emergency_luxury.dmm [hunk 2]
```diff
@@ -1,1326 +1,1830 @@
...
 /area/shuttle/escape/luxury)
-"bw" = (
-/obj/structure/window/reinforced{
-	dir = 1;
-	pixel_y = 2
+"Di" = (
+/obj/docking_port/mobile/emergency{
+	dir = 2;
+	name = "Luxurious Emergency Shuttle"
+	},
```
Comment (unit0016) [inline _maps/shuttles/emergency_luxury.dmm:644]: Also, one shuttlecode sufferer to another - you'll need to update `width` and `height` here seeing as the shuttle's larger than it was before. `dwidth` may need updating too. [Here](https://tgstation13.org/wiki/Guide_to_mapping#Shuttles)'s the best guide I can link to, but usually I just get strained and suffer with trial and error.

Comment (ZephyrTFA) [reply _maps/shuttles/emergency_luxury.dmm:644]: > Also, one shuttlecode sufferer to another - you'll need to update `width` and `height` here seeing as the shuttle's larger than it was before. `dwidth` may need updating too. [Here](https://tgstation13.org/wiki/Guide_to_mapping#Shuttles)'s the best guide I can link to, but usually I just get strained and suffer with trial and error.

No, do not do this. This is done automatically and should never have been done circa 2016

Comment (mc-oofert) [reply _maps/shuttles/emergency_luxury.dmm:644]: lmao ratio

**Comment by mc-oofert** [comment]
> Why remove shuttle rotation on the shutters instead of just... rotating the shutters? Would definitely shaft other shuttles where these are rotated, especially with wallening in the picture.

all my homies hate wallening and horizontal shutters are an eyesore

File: _maps/shuttles/emergency_luxury.dmm (hunk 2 above)
Comment (unit0016) [reply _maps/shuttles/emergency_luxury.dmm:644]: > No, do not do this. This is done automatically and should never have been done circa 2016

Really? Damn. I'm guessing that's just for emergency shuttles only then considering I've historically had to fuck with the vars on other shuttle types to get them to load.

Comment (ZephyrTFA) [reply _maps/shuttles/emergency_luxury.dmm:644]: > > No, do not do this. This is done automatically and should never have been done circa 2016
> 
> Really? Damn. I'm guessing that's just for emergency shuttles only then considering I've historically had to fuck with the vars on other shuttle types to get them to load.

No, it's for all shuttles.

Comment (unit0016) [reply _maps/shuttles/emergency_luxury.dmm:644]: MAN REALLY?
Since when, I *need* to backport to some other repos that so nobody else suffers, hell yeah.

**Comment by unit0016** [comment]
> all my homies hate wallening and horizontal shutters are an eyesore

I just feel that the stations widely using horizontal shutters everywhere while shuttles are hardcoded to not be able to is incredibly goofy.

File: _maps/shuttles/emergency_luxury.dmm (hunk 2 above)
Comment (ZephyrTFA) [reply _maps/shuttles/emergency_luxury.dmm:644]: #71220
#71687

**Comment by mc-oofert** [comment]
> > all my homies hate wallening and horizontal shutters are an eyesore
> 
> I just feel that the stations on /tg/ widely using horizontal shutters everywhere while shuttles are hardcoded to not be able to is incredibly goofy.

i have literally never seen a station with horizontal shutters

**Comment by unit0016** [comment]
> i have literally never seen a station with horizontal shutters
> 
> dont you mean blast doors?

Current /tg/ Deltastation Robotics:
image

**Comment by mc-oofert** [comment]
> > i have literally never seen a station with horizontal shutters
> > dont you mean blast doors?
> 
> Current /tg/ Deltastation Robotics: image
> 
> These were applied systematically to every map in #68056 and #68464 respectively.

imagine playing deltastation

**Review by MrMelbert — CHANGES_REQUESTED** [review]

File: code/modules/shuttle/shuttle_rotate.dm [hunk 3]
```diff
@@ -110,3 +110,8 @@ If ever any of these procs are useful for non-shuttles, rename it to proc/rotate
 	. = ..()
 	if(wall_turret_direction && (params & ROTATE_DIR))
 		wall_turret_direction = turn(wall_turret_direction,rotation)
+
+//shutters that go from the sides look bad
+/obj/machinery/door/poddoor/shutters/shuttleRotate(rotation, params)
+	params = NONE
+	return ..()
```
[the remark this answers is not in the document]
Comment (mc-oofert) [reply code/modules/shuttle/shuttle_rotate.dm:117]: im just going to remove it instead

**Comment by VladinXXV** [comment]
Maybe it just isn't showing on the screenshot or something, but you seem to have a lot of empty wall space and zero fire extinguishers. Might be a luxury worth adding? For the people with money, of course.

**Review by san7890 — CHANGES_REQUESTED** [review]
this is a cool concept designwise but this won't get it back into the server configuration as it doesn't really fulfill anything outlined here: https://tgstation13.org/phpBB/viewtopic.php?p=662250#p662250

If you were to make it emag-only, **I would be perfectly satisfied with it from the headmin standpoint**. I'm wondering if you could lean more into making it so that you didn't have to pay as much credits to get onboard, but all of the amenities would be marked up higher (i.e. you purchase caviar "on demand" for a steep price).  

Mappingwise, great job. I'll review that part as days go past, do remind me if it's been a bit. Feel free to mark it as "Ready for Review" when you address this in some capacity.

**Comment by blackdav123** [comment]
did you put fire extinguishers anywhere or am I just blind

**Comment by mc-oofert** [comment]
> this is a cool concept designwise but this won't get it back into the server configuration as it doesn't really fulfill anything outlined here: https://tgstation13.org/phpBB/viewtopic.php?p=662250#p662250
> 
> If you were to make it emag-only, **I would be perfectly satisfied with it from the headmin standpoint**. I'm wondering if you could lean more into making it so that you didn't have to pay as much credits to get onboard, but all of the amenities would be marked up higher (i.e. you purchase caviar "on demand" for a steep price).
> 
> Mappingwise, great job. I'll review that part as days go past, do remind me if it's been a bit. Feel free to mark it as "Ready for Review" when you address this in some capacity.

made it emag-only at 10000 credits now

**Comment by Draggeru** [comment]
Nameless mentioned a shooting range where the rich could shoot at the poor, I fully support this idea as well

**Comment by mc-oofert** [comment]
> Edit: Yes I see the guns. Make it an actual shooting range. Make the guns real guns or you're a coward

i dont think giving people guns for free is a good idea on a shuttle

**Comment by MrrFiish** [comment]
> > Edit: Yes I see the guns. Make it an actual shooting range. Make the guns real guns or you're a coward
> 
> i dont think giving people guns for free is a good idea on a shuttle

They ARE making it an emag shuttle, honestly I could see them being real.

**Comment by mc-oofert** [comment]
> > > Edit: Yes I see the guns. Make it an actual shooting range. Make the guns real guns or you're a coward
> > 
> > 
> > i dont think giving people guns for free is a good idea on a shuttle
> 
> They ARE making it an emag shuttle, honestly I could see them being real.

its an emag-only shuttle on the basis of being covered in fuck-off-indestructible walls

**Comment by Draggeru** [comment]
> > > > Edit: Yes I see the guns. Make it an actual shooting range. Make the guns real guns or you're a coward
> > > 
> > > 
> > > i dont think giving people guns for free is a good idea on a shuttle
> > 
> > 
> > They ARE making it an emag shuttle, honestly I could see them being real.
> 
> its an emag-only shuttle on the basis of being covered in fuck-off-indestructible walls

But we don't want the shuttle to be indestructible anymore.

**Comment by mc-oofert** [comment]
> Edit: My source of this being I talked to San about it. One of the largest grievances with the shuttle is BECAUSE its indestructible

image
the arena shuttle is indestructible too so

**Comment by san7890** [comment]
> But we don't want the shuttle to be indestructible anymore.
> 
> Edit: My source of this being I talked to San about it. One of the largest grievances with the shuttle is BECAUSE its indestructible

you're right in one way, and it's my fault you're wrong otherwise. the best case scenario would be making it not indestructible somehow, but the more I think on it and the more people who try and tackle this problem, the more I am convinced that there are two different spheres. Here's a diagram to illustrate:

image

A would be the concept behind the map: a great and silly critique of capitalism and other money-grinding tendencies, while rewarding certain player behaviors or being a good fuckin talk on society. i love the idea of people choking down caviar as they see the "poors" slum around a pigpen. It appeals to me.

However, the "B" Sphere is... everything else about this game. It's a sandbox game, anything that shows up on the station z-level should be able to be messed with somehow. Walls are meant to be broken, and you're supposed to be able to do anything you want in some way. If someone makes a five-foot tall wall, you get a six-foot long ladder through some game mechanic. One thing that I've really disliked about this shuttle's maintainability is that in order to keep the "A" sphere's message intact, you have to attach yourself to a complete disconnect to what the rest of the game should stand for (in my opinion). Some player finally cracks out a way to bypass it somehow? Well now you've got to wonder if their ingenuity should be rewarded, but also if it's a "bug" or "oversight" that _must_ be patched in order to create a more perfect codebase. It's.. blaugh.

However, they remain inextricably linked. I got sick of it back in August 2022 (see the forum post I linked earlier in this PR), and I got sick of people getting sick of it, so I removed it from the servers. Since that point, I've remained in hope that one of the twelve people who said they wanted to try and rework some aspect of it could finally find that perfect balance (the `????` zone in the diagram), but I think that as time goes on we'll never get that.

This is why I'm okay with it dropping down to emag-only status. I can't get everything I want in life, and no one really can. I just want to be less sick of it, and I want to be less sick of seeing it (this was the second most purchased emergency shuttle, excluding purpose-built station shuttles):

image

If it's on emag only status, it locks it behind some (hopefully) fun traitor interactions or a way to chain Crab-17 and the shuttle to troll the crew in a funny way (unless it becomes a complete bore due to overuse)... at least I know that someone who could have just as easily murdered the crew had some part in bringing about such a griefy shuttle. I want _more_ of course, and I think that you could re-adapt the entire concept to spin it on its head to bring about a much better concept that does a much better job of linking our "A" and "B" spheres... but I'm okay with it being emag only for now.

i hope that clarifies in a way that i poorly attempted to address in either forum post, speaking from a purely design perspective. I'm already interested in incorporating this new shuttle design (broadly, haven't reviewed it yet), I just desperately wanted _something_ in the status quo to change rather than be stuck in mediocrity.

**Comment by mc-oofert** [comment]
i have replaced the laser practice guns with real lasers that cost 20 credits each shot

File: code/modules/projectiles/guns/energy/laser.dm [hunk 4]
```diff
@@ -195,3 +195,14 @@
...
+
+// luxury shuttle funnies
+/obj/item/firing_pin/paywall/luxury
+	multi_payment = TRUE
+	owned = TRUE
+	payment_amount = 20
+
+/obj/item/gun/energy/laser/luxurypaywall
+	name = "luxurious laser gun"
+	desc = "A laser gun modified to cost 30 credits to fire. Point towards poor people."
```
Comment (san7890) [inline code/modules/projectiles/guns/energy/laser.dm:207]:
```suggestion
	desc = "A laser gun modified to cost 20 credits to fire. Point towards poor people."
```

**Comment by Redrover1760** [comment]
> i have replaced the laser practice guns with real lasers that cost 20 credits each shot

If I may suggest something, you should replace some of the walls of the poor medbay with windows, and maybe replace the southern cargo area with windows as well, to give the poors worse cover and less obvious places to hide if they are getting shot up from the Luxury side. 

And to allow the rich to laugh at the poor easier, without anywhere to hide from the scorn of being a poor person.

**Review by san7890 — CHANGES_REQUESTED** [review]
image

fix this weird floating light

**Comment by mc-oofert** [comment]
> fix this weird floating light

ok

**Pull request merged**